src/sorta.nim

```diff
@@ -409,45 +421,83 @@ proc `$`*[Key, Val](b: SortedTable[Key, Val]): string =
 
 proc len*[Key, Val](b: SortedTable[Key, Val]): int {.inline.} = b.entries
 
-
-iterator entries[Key, Val](b: SortedTable[Key, Val]): Entry[Key, Val] =
-  template populateWithLeftmostChildren(stack, currentNode) =
-    while not currentNode.isNil:
-      for i in countdown(currentNode.m-1, 0):
-        stack.add(currentNode.e[i])
-      currentNode = currentNode.p0
-
-  var
-    root = b.root
-    currentEntry: Entry[Key, Val]
-    stack = newSeq[Entry[Key, Val]]()
-
-  stack.populateWithLeftmostChildren(root)
-  while stack.len > 0:
-    currentEntry = stack.pop()
-    yield currentEntry
-    stack.populateWithLeftmostChildren(currentEntry.p)
-
+proc key[Key, Val](cursor: CursorPosition[Key, Val]): Key =
+  ## Return the key for a given cursor position
+  cursor.node.e[cursor.entry].key
+
+proc val[Key, Val](cursor: CursorPosition[Key, Val]): Val =
+  ## Return the value for a given cursor position
+  cursor.node.e[cursor.entry].val
+
+proc search[Key, Val](b: SortedTable[Key, Val], key: Key): Cursor[Key, Val] =
+  ## calculate the cursor pointing to the given key
+  var a = b.root
+  while not a.isNil:
+    var r = binarySearch(key, a)
+    if r < a.m:
+      result.stack.add((a, r))
+      if eq(key, a.e[r].key):
+        break
+    a = if r == 0: a.p0 else: a.e[r-1].p
+  # add a dummy entry for first next call
+  result.stack.add((nil, 0))
+
+proc current[Key, Val](cursor: Cursor[Key, Val]): CursorPosition[Key, Val] =
+  ## the current position of a cursor
+  last(cursor.stack)
```

Comment:
Isn't it easier to do just `cursor.stack[^1]` than to have and call this proc?

Reply:
Yes, I'm new to Nim and not yet so familiar with the slice notation. I will update as suggested. Thanks!